haystack/dataclasses/answer.py

```diff
@@ -9,6 +9,46 @@ class Answer:
     query: str
     metadata: Dict[str, Any]
 
+    def to_dict(self, flatten=True) -> Dict[str, Any]:
+        """
+        Converts Answer into a dictionary.
+
+        :param flatten: Whether to flatten `metadata` field or not. Defaults to `True`.
+        """
+        data = asdict(self)
+        if flatten:
+            meta = data.pop("metadata")
+            return {**data, **meta}
+        return data
+
+    @classmethod
+    def from_dict(cls, data_dict: Dict[str, Any]) -> "Answer":
+        """
+        Creates a new Answer object from a dictionary.
+        """
+        # Store metadata for a moment while we try un-flattening allegedly flatten metadata.
+        # We don't expect both a `metadata=` keyword and flatten metadata keys so we'll raise a
+        # ValueError later if this is the case.
+        meta = data_dict.pop("metadata", {})
```

Comment:
This code block (up until the end of the method) is the same in the implementation for `ExtractedAnswer` and `GeneratedAnswer`. We could refactor that to reduce code duplication.